lib/src/main/java/io/ably/lib/http/Http.java

```diff
@@ -209,18 +202,6 @@ public void finalize() {
 		dispose();
 	}
 
-	private static class HostFailedException extends AblyException {
-		private static final long serialVersionUID = 1L;
-		public HostFailedException(Throwable cause) {
-			super(cause);
-		}		
-		private static AblyException checkFor(Throwable t) {
-			if(t instanceof ConnectException || t instanceof UnknownHostException || t instanceof NoRouteToHostException)
```

Comment:
> I have moved the HostFailedException outside Http for it was forcing parent class AblyException constructor to be public.

That's fine.

> Also, I have moved the creation logic to single factory method for a class deciding whether itself or its super class going to be created seemed like an anti-pattern to me.

I understand that, but I don't like the way that `HostFailedException` is declared in one place, but the logic that decides what it is is somewhere else.

I would be happier perhaps with `HostFailedException` perhaps being a static inner class of `AblyException`.